listenbrainz/webserver/static/js/src/RecentListens.tsx

```diff
@@ -235,7 +235,15 @@ export default class RecentListens extends React.Component<
   };
 
   receiveNewListen = (newListen: string): void => {
-    const listen = formatWSMessageToListen(JSON.parse(newListen));
+    let json;
+    try {
+      json = JSON.parse(newListen);
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error("Coudn't parse the new listen as JSON: ", json);
```

Comment:
`json` is always going to be undefined in this context.

More useful would be to print the `error`  instead

Reply:
silly mistake :/